### PR TITLE
feat: text in stage

### LINKED ISSUE #939: Text support for stage channels
Stage channels now support messages, similar to voice channels.

While receiving messages works, most text channel features are missing (`Context.send` does work, though):
https://github.com/discord/discord-api-docs/commit/648109e7b7f88758e51fafb1a7f9fc6755b2b934

Additionally, there are new message types `27`, `28`, `29`, and `31`:
https://github.com/discord/discord-api-docs/commit/bf3c144bd5b486d565baed5c3db409e9ac8da61e

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

closes #939

## Checklist

- [ ] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [x] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by onerandomusername** [comment]
This is now ready for review, but I think I need to separate some of the changes into another pull, I think I squashed a few too many bugs to include them all in this pull.

**Review by shiftinv — COMMENTED** [review]
`StageChannel` missing in docs in a few places:
- `InteractionMessage.channel`
- `Message.channel`
- `abc.Messageable`
- `PartialMessage` (x2)
- `on_typing` (x2)
- `AuditLogDiff.slowmode_delay`, `.user_limit`, `.nsfw`

File: disnake/guild.py [hunk 1]
```diff
@@ -1466,9 +1466,13 @@ async def create_stage_channel(
         topic: Optional[str] = MISSING,
         position: int = MISSING,
         bitrate: int = MISSING,
+        user_limit: int = MISSING,
+        rtc_region: Optional[Union[str, VoiceRegion]] = MISSING,
+        video_quality_mode: VideoQualityMode = MISSING,
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/guild.py:1471]: It is there, just wrong version, thanks!

File: disnake/interactions/base.py [hunk 2]
```diff
@@ -1420,7 +1420,7 @@ class InteractionMessage(Message):
         The actual contents of the message.
     embeds: List[:class:`Embed`]
         A list of embeds the message has.
-    channel: Union[:class:`TextChannel`, :class:`VoiceChannel`, :class:`Thread`, :class:`DMChannel`, :class:`GroupChannel`, :class:`PartialMessageable`]
+    channel: Union[:class:`TextChannel`, :class:`VoiceChannel`. :class:`StageChannel`, :class:`Thread`, :class:`DMChannel`, :class:`GroupChannel`, :class:`PartialMessageable`]
```
Comment (shiftinv) [inline disnake/interactions/base.py:1423]:
```suggestion
    channel: Union[:class:`TextChannel`, :class:`VoiceChannel`, :class:`StageChannel`, :class:`Thread`, :class:`DMChannel`, :class:`GroupChannel`, :class:`PartialMessageable`]
```

**Review by shiftinv — APPROVED** [review]

**Pull request merged**